[Bug]: panic: value is null

### Terraform Core Version

1.5.0

### AWS Provider Version

5.21.0

### Affected Resource(s)

I'm not sure

### Expected Behavior

Don't crash

### Actual Behavior

a panic during terraform apply

### Relevant Error/Panic Output Snippet

```shell
Stack trace from the terraform-provider-aws_v5.21.0_x5 plugin:

panic: value is null

goroutine 1584 [running]:
github.com/hashicorp/go-cty/cty.Value.LengthInt({{{0xf0cf538?, 0xc002b25160?}}, {0x0?, 0x0?}})
	github.com/hashicorp/go-cty@v1.4.1-0.20200414143053-d3edf31b6320/cty/value_ops.go:989 +0x27d
github.com/hashicorp/go-cty/cty.Value.AsValueMap({{{0xf0cf538?, 0xc002b25160?}}, {0x0?, 0x0?}})
	github.com/hashicorp/go-cty@v1.4.1-0.20200414143053-d3edf31b6320/cty/value_ops.go:1237 +0x71
github.com/hashicorp/terraform-provider-aws/internal/provider.tagsUpdateFunc({0xf0ce1c0, 0xc00336b080}, {0xf0d9ee8, 0xc0034cb800}, {0xf0d0a40, 0x155daf10}, 0xc00069b580, {0xc00017a215, 0x18}, {0xdd4c0ab, ...}, ...)
	github.com/hashicorp/terraform-provider-aws/internal/provider/tags_interceptor.go:59 +0x5ea
github.com/hashicorp/terraform-provider-aws/internal/provider.tagsResourceInterceptor.run({0xc00069b580?, 0xdff3b58?, 0xdff3b50?}, {0xf0ce1c0, 0xc00336b080}, {0xf0d9ee8, 0xc0034cb800}, {0xdce7f00?, 0xc00350efc0?}, 0x8, ...)
	github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:366 +0xbd7
github.com/hashicorp/terraform-provider-aws/internal/provider.interceptedHandler[...].func1(0x0?, {0xdce7f00?, 0xc00350efc0?})
	github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:127 +0x4e8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xf0ce1c0?, {0xf0ce1c0?, 0xc002cf75f0?}, 0xd?, {0xdce7f00?, 0xc00350efc0?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:807 +0x87
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000bc8fc0, {0xf0ce1c0, 0xc002cf75f0}, 0xc003505520, 0xc0034cb680, {0xdce7f00, 0xc00350efc0})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:919 +0x845
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc001bb9a28, {0xf0ce1c0?, 0xc002cf74d0?}, 0xc002de3450)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/grpc_provider.go:1060 +0xe8d
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0xf0ce118?, {0xf0ce1c0?, 0xc002cf7110?}, 0xc002de3450)
	github.com/hashicorp/terraform-plugin-mux@v0.12.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x1b5
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc001bf12c0, {0xf0ce1c0?, 0xc002cf65d0?}, 0xc001c79ab0)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/tf5server/server.go:859 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xda6dfe0?, 0xc001bf12c0}, {0xf0ce1c0, 0xc002cf65d0}, 0xc001c79a40, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:467 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000052f00, {0xf0db9e0, 0xc0005221a0}, 0xc001fad7a0, 0xc003531d70, 0x15581a58, 0x0)
	google.golang.org/grpc@v1.57.0/server.go:1360 +0xe23
google.golang.org/grpc.(*Server).handleStream(0xc000052f00, {0xf0db9e0, 0xc0005221a0}, 0xc001fad7a0, 0x0)
	google.golang.org/grpc@v1.57.0/server.go:1737 +0xa36
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.57.0/server.go:982 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.57.0/server.go:980 +0x18c

Error: The terraform-provider-aws_v5.21.0_x5 plugin crashed!
```


### Terraform Configuration Files

a lot. I don't know what is relevant.

### Steps to Reproduce

I don't know. it happened during terraform apply once.

### Debug Output

_No response_

### Panic Output

panic: value is null

goroutine 1584 [running]:
github.com/hashicorp/go-cty/cty.Value.LengthInt({{{0xf0cf538?, 0xc002b25160?}}, {0x0?, 0x0?}})
	github.com/hashicorp/go-cty@v1.4.1-0.20200414143053-d3edf31b6320/cty/value_ops.go:989 +0x27d
github.com/hashicorp/go-cty/cty.Value.AsValueMap({{{0xf0cf538?, 0xc002b25160?}}, {0x0?, 0x0?}})
	github.com/hashicorp/go-cty@v1.4.1-0.20200414143053-d3edf31b6320/cty/value_ops.go:1237 +0x71
github.com/hashicorp/terraform-provider-aws/internal/provider.tagsUpdateFunc({0xf0ce1c0, 0xc00336b080}, {0xf0d9ee8, 0xc0034cb800}, {0xf0d0a40, 0x155daf10}, 0xc00069b580, {0xc00017a215, 0x18}, {0xdd4c0ab, ...}, ...)
	github.com/hashicorp/terraform-provider-aws/internal/provider/tags_interceptor.go:59 +0x5ea
github.com/hashicorp/terraform-provider-aws/internal/provider.tagsResourceInterceptor.run({0xc00069b580?, 0xdff3b58?, 0xdff3b50?}, {0xf0ce1c0, 0xc00336b080}, {0xf0d9ee8, 0xc0034cb800}, {0xdce7f00?, 0xc00350efc0?}, 0x8, ...)
	github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:366 +0xbd7
github.com/hashicorp/terraform-provider-aws/internal/provider.interceptedHandler[...].func1(0x0?, {0xdce7f00?, 0xc00350efc0?})
	github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:127 +0x4e8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xf0ce1c0?, {0xf0ce1c0?, 0xc002cf75f0?}, 0xd?, {0xdce7f00?, 0xc00350efc0?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:807 +0x87
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000bc8fc0, {0xf0ce1c0, 0xc002cf75f0}, 0xc003505520, 0xc0034cb680, {0xdce7f00, 0xc00350efc0})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:919 +0x845
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc001bb9a28, {0xf0ce1c0?, 0xc002cf74d0?}, 0xc002de3450)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/grpc_provider.go:1060 +0xe8d
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0xf0ce118?, {0xf0ce1c0?, 0xc002cf7110?}, 0xc002de3450)
	github.com/hashicorp/terraform-plugin-mux@v0.12.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x1b5
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc001bf12c0, {0xf0ce1c0?, 0xc002cf65d0?}, 0xc001c79ab0)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/tf5server/server.go:859 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xda6dfe0?, 0xc001bf12c0}, {0xf0ce1c0, 0xc002cf65d0}, 0xc001c79a40, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:467 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000052f00, {0xf0db9e0, 0xc0005221a0}, 0xc001fad7a0, 0xc003531d70, 0x15581a58, 0x0)
	google.golang.org/grpc@v1.57.0/server.go:1360 +0xe23
google.golang.org/grpc.(*Server).handleStream(0xc000052f00, {0xf0db9e0, 0xc0005221a0}, 0xc001fad7a0, 0x0)
	google.golang.org/grpc@v1.57.0/server.go:1737 +0xa36
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.57.0/server.go:982 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.57.0/server.go:980 +0x18c

### Important Factoids

_No response_

### References

_No response_

### Would you like to implement a fix?

No

--- a/.changelog/34073.txt
+++ b/.changelog/34073.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+provider/tags: Prevent crash when `tags_all` is null
+```

--- a/internal/provider/tags_interceptor.go
+++ b/internal/provider/tags_interceptor.go
@@ -56,8 +56,12 @@ func tagsUpdateFunc(ctx context.Context, d schemaResourceData, sp conns.ServiceP
 	stateTags := make(map[string]string)
 	if state := d.GetRawState(); !state.IsNull() && state.IsKnown() {
 		s := state.GetAttr(names.AttrTagsAll)
-		for k, v := range s.AsValueMap() {
-			stateTags[k] = v.AsString()
+		if !s.IsNull() {
+			for k, v := range s.AsValueMap() {
+				if !v.IsNull() {
+					stateTags[k] = v.AsString()
+				}
+			}
 		}
 	}
 

